drivers/src/nrfx_nvmc.c

```diff
@@ -189,27 +189,42 @@ static uint32_t partial_word_create(uint32_t addr, uint8_t const * bytes, uint32
 
 static void nvmc_readonly_mode_set(void)
 {
-#if defined(NRF_TRUSTZONE_NONSECURE)
+    /*
+     * For secure code, the access mode needs to be set for both secure and
+     * non-secure regions.
+     */
+#if defined(NVMC_CONFIGNS_WEN_Msk)
```

Comment:
This #if is defined, regardless of wheather we are building a secure or a non-secure image